torch/csrc/jit/operator_upgraders/README.md

```diff
@@ -0,0 +1,95 @@
+# Guidance for Operator Developer
+
+After you change either the operator signature in a BC breaking way or the semantics of the operator, you will need to write an “upgrader” to make the change non-BC breaking iff they are used in TorchScript or mobile. In general, you can know your operator is BC breaking, if it fails `test/forward_backward_compatibility/check_forward_backward_compatibility.py `
+
+The steps to write upgrader:
+
+1. Prepare a model before making changes to the operator, following the process:
+     1. add a module in `test/jit/fixtures_srcs/fixtures_src.py`. The reason why switching to commit is that, an old model with the old operator before the change is needed to ensure the upgrader is working as expected. In `test/jit/fixtures_srcs/generate_models.py`, add the module and its corresponding changed operator like following
+
+        ```
+        ALL_MODULES = {
+           TestVersionedLinspaceV7(): "aten::linspace",
+        }
+        ```
+
+        This module should include the changed operator. If the operator isn't covered in the model, the model export process will fail.
+
+       2. Export the model to `test/jit/fixtures` by running
+        ```
+        python test/jit/fixtures_src/generate_models.py
+        ```
+
+       3. Commit the change and submit a pr
+
+2. Make changes to the operator and write an upgrader.
+    1. Make the operator change.
+    2. Write an upgrader in `caffe2/torch/csrc/jit/operator_upgraders/upgraders_entry.cpp` file inside a map `kUpgradersEntryMap`. The softly enforced naming format is `&lt;operator_name>_&lt;operator_overload>_&lt;start>_&lt;end>`. For example, the below example means that `linspace.out `and` linspace` at versions from 0 to 7 need to be replaced by these upgraders.
+
+    ```
+    static std::unordered_map<std::string, std::string> kUpgradersEntryMap(
+        {
+          {"linspace_0_7", R"SCRIPT(
+    def linspace_0_7(start: Union[int, float, complex], end: Union[int, float, complex], steps: Optional[int], *, dtype: Optional[int], layout: Optional[int],
+                     device: Optional[Device], pin_memory: Optional[bool]):
+      if (steps is None):
+        return torch.linspace(start=start, end=end, steps=100, dtype=dtype, layout=layout, device=device, pin_memory=pin_memory)
+      return torch.linspace(start=start, end=end, steps=steps, dtype=dtype, layout=layout, device=device, pin_memory=pin_memory)
+    )SCRIPT"},
+        }
+    ```
+
+    2. Bump the `kProducedFileFormatVersion` by 1 and provide the reasons under `caffe2/caffe2/versions.h `
```

Comment:
I think you also need to bump `kMaxProducedFileFormatVersion`

Reply:
Resolve